### PR TITLE
Render schemas at compile time

### PR DESCRIPTION
**Pull request by imbstack**
This code is ugly as sin, but I'm tired of looking at it. I _think_ it handles just about every case of json-schema. I'm sure there are some things I've left out, but it seems to get pretty close. This can't be landed until taskcluster/taskcluster-login#49 is landed, but then it should Just Work™.

If we'd like this code to be nicer I can circle back to this soon and do a cleanup pass. Ping me if you'd like an ngrok link.

### CODE REVIEW

**Comment by jonasfj** [comment]
Love it!

File: lib/raw.js [hunk 1]
```diff
@@ -59,22 +64,55 @@ module.exports = function() {
...
-      dat.marked = marked;
-      file.contents = new Buffer(refTemplate(dat));
-      file.inline = true;
-      file.data.docson = true;
-    }
 
-    this.push(file);
-    return cb();
+      // TODO: Get this all cleaned up
+      let promises = _.flatMap(dat.entries, ref => {
```
Comment (djmitche) [inline lib/raw.js]: this is shadowing the outer promises

File: lib/raw.js [hunk 2]
```diff
@@ -59,22 +64,55 @@ module.exports = function() {
...
+            }).then(s => {
+              ref[type] = s;
+            });
+          }
+          return Promise.resolve();
+        });
+      });
+
+      Promise.all(promises).then(() => {
+        dat.marked = marked;
```
Comment (djmitche) [inline lib/raw.js]: `dat` is not local to each promise..

**Comment by djmitche** [comment]
Oh, and one other nice thing to have here would be fetching references with https.  I think the docson stuff had `http` pretty deeply embedded in it.  Maybe that already works -- I don't really remember the details.  But it would make infosec happy if we turned of http for references and schemas.

**Comment by imbstack** [comment]
This actually doesn't fetch any references over the network at all. They're entirely transmogrified into local references to the schemas uploaded by lib-docs. I'll fix the other points as well!

**Comment by djmitche** [comment]
🤦‍

**Pull request merged**